### PR TITLE
enable query reduce on querynodes for shortcut return(#27306)

### PR DESCRIPTION
**Pull request by MrPresent-Han**
related: #27306

### CODE REVIEW

**Comment by MrPresent-Han** [comment]
rerun ut

File: internal/querynodev2/segments/result_test.go [hunk 1]
```diff
@@ -416,7 +416,7 @@ func (suite *ResultSuite) TestResult_MergeInternalRetrieveResults() {
 				FieldsData: []*schemapb.FieldData{fieldData},
 			}
 
-			_, err := MergeInternalRetrieveResult(context.Background(), []*internalpb.RetrieveResults{result},
+			_, err := MergeInternalRetrieveResult(context.Background(), []*internalpb.RetrieveResults{result, result},
```
Comment (yah01) [inline internal/querynodev2/segments/result_test.go:419]: what are the double results?

**Pull request merged**